qemu/tests/balloon_hotplug.py

```diff
@@ -88,18 +88,25 @@ def enable_balloon_service():
         devs = vm.devices.get_by_params({"id": 'balloon%d' % idx})
         vm.params["balloon_pci_bus"] = devs[0]["bus"]
 
+        if i == 0:
+            if params['os_type'] == 'windows':
+                balloon_test = BallooningTestWin(test, params, env)
+            else:
+                balloon_test = BallooningTestLinux(test, params, env)
+            min_sz, max_sz = balloon_test.get_memory_boundary()
+
```

Comment:
How about move l92-l96 out of 'for loop',and then no need if condition.

Reply:
We can't, to initialize balloon_test, balloon device must be hotpluged first

Reply:
I would recommend removing the `if i == 0`, better initialize it again(so `get_ballooned_memory` and probably `balloon_memory` again) every time plug the balloon device, to ensure the everything correct during the repeat process.

Reply:
Both are okay to me, initialize again will takes another 3min*100 = 5 hours waiting time.

Reply:
If initialize it every time, what's the point to replace balloon_check.run ?

Reply:
We need balloon_test in enable_balloon_service() function, if we do not replace balloon_check.run(), then we need to initialize balloon_test twice(one for enable_balloon_service, another for balloon_check.run)

Reply:
In summary:
initialize everytime and replace balloon_check.run() will save 5 hours;
initialize only for the first time will save 10 hours.